keps/3122-expose-flavors-in-localqueue-status/README.md

```diff
@@ -0,0 +1,154 @@
+# KEP-3122: Expose Flavors in LocalQueue Status
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Implementation overview](#implementation-overview)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+This KEP introduces a new status field in LocalQueue, allowing users to see
+all currently available ResourceFlavors in the LocalQueue.
+
+## Motivation
+
+Currently, users without RBAC access to ResourceFlavors cannot view the list
+of available flavors. Depending on the RBAC rules, users might also lack read
+access to ClusterQueues. Providing users with information about the available
+flavors is useful, as it gives them an idea of the capabilities provided by
+a LocalQueue (e.g., a flavor might include newer GPUs).
+
+### Goals
+
+- Provide a possibility to see all currently available ResourceFlavors in 
+  the LocalQueue.
+
+### Non-Goals
+
+- Verify that the ResourceFlavors exist and show only the existing flavors.
+
+## Proposal
+
+Introduce a new status field `availableFlavors` in LocalQueue 
+that will be updated when ClusterQueue flavors are modified.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user I want to see the list of all ResourceFlavors available in each LocalQueue 
+due to the RBAC configuration for ClusterQueue in my cluster I cannot inspect the 
+ClusterQueue objects directly, only LocalQueues.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+### API
+
+Create `AvailableFlavor` API object:
+
+```go
+type AvailableFlavor struct {
```

Comment:
What about the resources in each flavor?

Reply:
This hasn't come up directly yet, but it would be nice for users to be able to construct queries to make sure that whatever local queue they are submitting to has the necessary resources for the workload they are submitting to it

Reply:
In that case, we can expose the list of resources with the `coveredResources: []string` field, but without the quotas. if we don't see a usecase for quotas I would defer for later, maybe as opt-in, because I'm not sure if this would leak too much information to the users. WDYT?

Reply:
WDYT @KPostOffice?

Reply:
quotas would be tricky, because they might not be fully available to those users, so the information could just be confusing.

Reply:
yeah, moreover, the user may have even more quota than would be shown due to borrowing :)

Reply:
Ok, I don't see any downsides of including the resource names, and it may help for some likely scenarios. 

So I would suggest updating the KEP to include them, I think simply `resources []string` should be enough.

Reply:
This sounds great to me thank you all

Reply:
Done